### PR TITLE
[draft] introduce training backend

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit introduces training backend.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

File: runtime/onert/core/include/exec/TrainerSequence.h [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  /**
+   * @brief Call this function to initialize vars before running
+   * @note When we run a model with static tensor input and then run with dynamic tensor input,
+   *       _enable_dynamic_shape_inferer is set to @c false at first run.
+   *       Once _enable_dynamic_shape_inferer is set to @c true it cannot be changed to @c false
+   *       only with calling enableDynamicShapeInferer(). So initializing it to @c false is
+   *       necessary.
+   * @todo This is a quick fix. Adding this will increase time for run(). Find way to optimize.
+   */
+  void initRunning() { _enable_dynamic_shape_inferer = false; }
```
Comment (ragmani) [inline runtime/onert/core/include/exec/TrainerSequence.h:116]: ditto
```suggestion
```

File: runtime/onert/core/include/exec/TrainerSequence.h [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+protected:
+  std::vector<std::unique_ptr<ITrainerFunction>> _functions;
+
+protected:
+  bool _enable_dynamic_shape_inferer = false;
+
+  std::shared_ptr<DynamicTensorCtx> _dynamic_tensor_ctx = nullptr;
```
Comment (ragmani) [inline runtime/onert/core/include/exec/TrainerSequence.h:124]: ditto
```suggestion
```

**Review by ragmani — COMMENTED** [review]
`TrainerSequence` looks good.

**Comment by jyoungyun** [comment]
The detail implementation was discussed in Draft #10825.

**Pull request closed without merging**